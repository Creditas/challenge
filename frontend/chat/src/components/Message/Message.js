import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import styles from './message.css';

const Message = ({text}) => (
  <div 
    className={styles.base}
    ref={el => el && el.scrollIntoView({ behavior: "smooth" })}
  >
    <Paper>
      <div className={styles.msgWrapper}>
        {text}
      </div>
    </Paper>
  </div>
);

Message.propTypes = {
  text: PropTypes.string.isRequired
}

export default Message;
