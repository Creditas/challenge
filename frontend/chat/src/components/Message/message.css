.base {
  margin-top: 20px;
}

.msgWrapper {
  padding: 10px 20px;
}
