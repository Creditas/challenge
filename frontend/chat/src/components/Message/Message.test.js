import React from 'react';
import { shallow } from 'enzyme';
import Message from './Message';

describe('Message', () => {
  it('should render', () => {
    const rendered = shallow(<Message text="Mock" />);
    expect(rendered).toMatchSnapshot()
  })
})
