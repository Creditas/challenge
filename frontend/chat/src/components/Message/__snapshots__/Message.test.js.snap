// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message
    text="Mock"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Paper)>
        <div
                className="msgWrapper"
        >
                Mock
        </div>
</WithStyles(Paper)>,
      "className": "base",
    },
    "ref": [Function],
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="msgWrapper"
>
          Mock
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Mock",
          "className": "msgWrapper",
        },
        "ref": null,
        "rendered": "Mock",
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Paper)>
          <div
                    className="msgWrapper"
          >
                    Mock
          </div>
</WithStyles(Paper)>,
        "className": "base",
      },
      "ref": [Function],
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="msgWrapper"
>
            Mock
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Mock",
            "className": "msgWrapper",
          },
          "ref": null,
          "rendered": "Mock",
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
